.github/workflows/pythonbuild.yml

```diff
@@ -35,7 +35,15 @@ jobs:
         python -m pip install --upgrade pip setuptools wheel
         make setup${{ matrix.spark-version-suffix }}
         make -C plugins install-all-dev
+        sudo bash -c 'curl -L https://github.com/dolthub/dolt/releases/latest/download/install.sh | sudo bash'
         pip freeze
+    - name: Install Dolt
```

Comment:
Can we remove this?  Can we make this part of the plugin test?

Reply:
This is the "which database to test with" trade-off. Testing with Dolt was easy but I need this binary. I think this should be replaced by more comprehensive tests with Postgres and Mysql, but those would have a bigger testing footprint and a longer dev-timeline.

Reply:
maybe the plugin could be tested with SQLite, which wouldn't have dependency issues, but that might take me a bit to figure out

Reply:
You're right that we should run plugin unit tests as part of CI but that gets kicked off from the [Makefile](https://github.com/flyteorg/flytekit/blob/master/Makefile#L39) which depends on this [setup](https://github.com/flyteorg/flytekit/blob/master/.github/workflows/pythonbuild.yml#L37) line which then runs this pip [install](https://github.com/flyteorg/flytekit/blob/master/plugins/Makefile#L7) so the sqlalchemy setup.py is still run.  Any setup steps you put there will be executed.  It sounds like the implication is that pip installing this plugin isn't quite enough to get started, there's a few more commands to run after the pip install step before the user can start using it?  The [constructs around post-install](https://stackoverflow.com/questions/20288711/post-install-script-with-python-setuptools) hooks seem scarce though.  Can we replace with default behavior by any chance?

I don't know much about sqlalchemy...  is this plugin for all sqlalchemy?  Or is it only for dolt?  If there are hard dependencies on dolt for this plugin, should we rename to DoltSqlAlchemy?

Sorry for all the back and forth!  I'm out most of tomorrow (tuesday) but can do a google meet in the morning or the day after if maybe that's faster?  And thank you for doing this!

Reply:
no worries! [this doc](https://docs.google.com/document/d/147nNlWrEJz7zLfIrr4rkokUceP8wFiCJuxIm2sS8ZZs/edit?usp=sharing) is my summary for what sqlalchemy is as a short summary. feel free to ping me in slack or discord to meet whenever. will follow up with specific QA below.

Reply:
sqlalchemy is generic, not just for dolt. you put any SQL server string in there, specify the dialect (mysql, postgres, etc) and it will issue the proper select syntax. if you look at the code in  PR summary above, it's quite concise -- a connect string and a pandas read.

Reply:
Setting up test database servers for every example data source is much more difficult. Each database server requires a different package except sqlite, maybe. That's why I added dolt as a test dependency -- to make a database server test fixture. I could have used postgres, mysql, oracle or sqlite instead.

Reply:
I'm quickest with dolt -- in retrospect sqlite might have been a better choice. Ketan said you guys would have a way to inject server dependencies soon, so I assumed my tests would be phased out.

Reply:
Post-install hooks would include the test dependencies in the package, which I think you want to avoid.

Reply:
Thanks for taking so much time looking at this! I got the initial version working in 20 minutes but I really should have used sqlite -- saved us both from the back and forth